activation/nipost.go

```diff
@@ -81,12 +86,15 @@ type NIPostBuilder struct {
 	minerID           []byte
 	db                *sql.Database
 	postSetupProvider PostSetupProvider
-	poetProver        PoetProvingServiceClient
+	poetProvers       []PoetProvingServiceClient
 	poetDB            poetDbAPI
 	state             *BuilderState
 	log               log.Log
 }
 
+// A compile time check to ensure that `NIPostBuilder` fully implements the `nipostBuilder` interface.
+var _ nipostBuilder = (*NIPostBuilder)(nil)
```

Comment:
Not required. `NIPostBuilder` doesn't define the `nipostBuilder` interface, but its user `activation.Builder` does.

Even without this line the compiler would still complain that `NIPostBuilder` doesn't implement the `nipostBuilder` interface if it wouldn't.

Reply:
It also serves as a note that `NIPostBuilder` is supposed to implement `nipostBuilder`. This note sits next to the definition of the struct. IMO it is useful.

Reply:
Removed